### PR TITLE
Building process improvement

### PR DESCRIPTION
**Pull request by omgovich**
Hey, @RyanChristian4427 👋 Check this out.

I took your concept from #25, but keeps packages ignored by Git.

Also, I made the building process parallel by `Promise.all(...)`

### CODE REVIEW

**Comment by RyanChristian4427** [comment]
Looks good. We might have to inject the `types` field after the fact though. Just noticed my solution, when merged into the TS branch had a pretty big failing because of this. Little bit of awkwardness with Microbundle.

File: build-packages.js [hunk 1]
```diff
@@ -2,48 +2,73 @@ const fs = require("fs");
...
+      await makeDir(outputDirPath);
       await writeFile(manifestPath, manifestCode, "utf8");
     }
-  }
+
+    // Bundler options
+    const args = {
+      name: packageName,
+      cwd: isMainPackage ? __dirname : outputDirPath,
+      output: `${outputDirPath}/index.js`,
```
Comment (RyanChristian4427) [inline build-packages.js:54]:
```suggestion
      output: `${outputDirPath}`,
```

I don't think you need the `.js` here. [Microbundle's `output` is for a directory](https://github.com/developit/microbundle/#all-cli-options-)

Comment (RyanChristian4427) [reply build-packages.js:54]: Whoops, didn't catch you in time. Sorry.

File: build-packages.js [hunk 2]
```diff
@@ -2,48 +2,73 @@ const fs = require("fs");
...
       // Create `package.json`
       var manifestCode = JSON.stringify({
-        name: `react-colorful-${name}`,
+        name: packageName,
         private: true,
         main: "index.js",
         module: "index.module.js",
         esmodule: "index.esmodule.js",
-        types: "../types/index.d.ts"
+        "umd:main": "dist/index.umd.js",
```
Comment (RyanChristian4427) [inline build-packages.js:40]: Oh, and this doesn't look correct. Sorry, didn't catch it.

File: build-packages.js (hunk 1 above)
Comment (omgovich) [reply build-packages.js:54]: Don't know how it works, but If I remove `index.js` microbundle names the files like that:
image

Comment (omgovich) [reply build-packages.js:54]: Looks like Microbundle doesn't look at `main: "index.js"` in `package.json`. Hmmm

Comment (omgovich) [reply build-packages.js:54]: They use a file name in their own examples, so I think it's okay:
https://github.com/developit/microbundle/#building-a-single-bundle-with-a-fixed-output-name

Comment (RyanChristian4427) [reply build-packages.js:54]: That's really odd. I don't think I've seen it doing that before. 

Though as I usually stick to TS, I guess I've avoided that issue. Nvm then. Don't know how that happens.

File: build-packages.js (hunk 2 above)
Comment (omgovich) [reply build-packages.js:40]: Not sure that I got what you mean)

Comment (RyanChristian4427) [reply build-packages.js:40]: The `dist`. If you try to use the `umd:main` you'll be pointed at a file that doesn't exist. Should be `index.umd.js`.

Comment (RyanChristian4427) [reply build-packages.js:40]: Microbundle still generates the right file, it's just the path that's incorrect.

Comment (omgovich) [reply build-packages.js:40]: Oh, I see. Nice catch, thanks!

**Pull request merged**